main/conversion.ts

```diff
@@ -1,83 +1,43 @@
 import fs from 'fs';
-import {app, clipboard, dialog} from 'electron';
+import {app, clipboard} from 'electron';
 import {EventEmitter} from 'events';
-import {ConversionOptions, ConversionStatus, CreateConversionOptions, Format} from './common/types';
+import {ConversionOptions, Format} from './common/types';
 import {Video} from './video';
 import {convertTo} from './converters';
-import Export, {ExportOptions} from './export';
 import hash from 'object-hash';
-import {ipcMain as ipc} from 'electron-better-ipc';
-import {plugins} from './plugins';
-import {askForTargetFilePath} from './plugins/built-in/save-file-plugin';
-import path from 'path';
-import {showError} from './utils/errors';
 import {notify} from './utils/notifications';
 import PCancelable from 'p-cancelable';
 import prettyBytes from 'pretty-bytes';
-import {ensureDockIsShowingSync} from './utils/dock';
-import {windowManager} from './windows/manager';
+import TypedEventEmitter from 'typed-emitter';
 
 const plist = require('plist');
 
-// TODO: remove this when exports window is rewritten
-const callExportsWindow = (channel: string, data: any) => {
-  const exportsWindow = windowManager.exports?.get();
-
-  if (exportsWindow) {
-    // TODO(karaggeorge): Investigate why `ipc.callRenderer(exportsWindow, channel, data);` is not working here.
-    ipc.callRenderer(exportsWindow, channel, data);
-  }
-};
-
 // A conversion object describes the process of converting a video or recording
 // using ffmpeg that can then be shared multiple times using Share plugins
-export default class Conversion extends EventEmitter {
+export default class Conversion extends (EventEmitter as new () => TypedEventEmitter<ConversionEvents>) {
   static conversionMap = new Map<string, Conversion>();
 
   static get all() {
     return [...this.conversionMap.values()];
   }
 
-  id: string;
-  video: Video;
-  format: Format;
-  options: ConversionOptions;
-
-  text = '';
-  percentage?: number;
-  error?: Error;
-  description: string;
-  title: string;
+  readonly id: string;
   finalSize?: string;
   convertedFilePath?: string;
-  requestedFileType?: Format;
-
-  private currentExport?: Export;
-  private _status: ConversionStatus = ConversionStatus.idle;
-
-  get status() {
-    return this._status;
-  }
-
-  set status(newStatus: ConversionStatus) {
-    this._status = newStatus;
-    this.emit('updated');
-  }
 
   get canCopy() {
     return Boolean(this.convertedFilePath && [Format.gif, Format.apng].includes(this.format));
   }
 
   private conversionProcess?: PCancelable<string>;
 
-  constructor(video: Video, format: Format, options: ConversionOptions) {
+  constructor(
+    public readonly video: Video,
+    public readonly format: Format,
+    public readonly options: ConversionOptions
+  ) {
+    // eslint-disable-next-line constructor-super
```

Comment:
This is to silence a warning because of the TypedEmitter annotation. Couldn't find a way around it.

Warning said this doesn't need a `super()` call, but obviously without one we get a compilation error